### DISCUSSION TITLE
Controller  the "as"  element attributes

### DISCUSSION THREAD

**Discussion by andrconstruction** [react-hook-form/react-hook-form#1896] [Q&A] [answered]
How can I change some attributes or a class on the **as** element on errors or focus, for example I want to set the class 'is-valid'  if it is Bootstrap FormControl?

**Comment by bluebill1049** [comment, marked as answer]
`as={<xxxx is-valid='xxx' />}`

**Reply by andrconstruction** [reply]
Can  you provide an example, please?

**Reply by bluebill1049** [reply]
sorry take a look again, it was hidden before.

**Reply by andrconstruction** [reply]
<Form.Group>
                <Form.Label>Логин</Form.Label>
                <Controller
                    as={Form.Control}
                    rules={{
                        required : "обязательное поле"
                        , maxLength : 255, minLength: 2 }}
                    name="username"
                    control={control}
                    defaultValue={userData.user.username}
                />
                <ErrorMessage
                    errors={errors}
                    style={{ color: "red" }}
                    as={"p"}
                    name="username"
                    render={({ messages }) =>
                        messages &&
                        Object.entries(messages).map(([type, message]) => (
                            <p key={type} style={{ color: "red" }}>{message}</p>
                        ))
                    }
                />
            </Form.Group>

I meant that if i got an error, should I make a function for attribute and how it will looks/
How to add/remove this attributes from a validsation function or on change function, how to link to an element?

**Reply by andrconstruction** [reply]
Or better should I use useState for it

**Reply by bluebill1049** [reply]
no you should leverage `errors` object.

**Reply by andrconstruction** [reply]
is errors contains a reffrerence on the as object?

**Reply by andrconstruction** [reply]
Can you give a little example within the sandbox

**Reply by andrconstruction** [reply]
I tryed to make next

<Controller
                    as={<FormControl isInvalid={errors && errors.username} /> }
                    rules={{
                        required : "обязательное поле"
                        , maxLength : 255, minLength : 2
                    }}
                    name="username"
                    control={control}
                    defaultValue={userData.user.username}
                />

But got an error  :
ncaught Error: Objects are not valid as a React child (found: object with keys {type, message, ref}). 

What do I wrong?

**Reply by andrconstruction** [reply]
Huraaaa its works!!!!
All I did in last tyr was  correct. Error was in other element
